addons/account_peppol/tools/demo_utils.py

```diff
@@ -143,8 +143,11 @@ def handle_demo(func, self, *args, **kwargs):
     """
     def get_demo_mode_account_edi_proxy_client_user(self, args, kwargs):
         if self.id:
-            return self.edi_mode == 'demo'
-        return self.env['ir.config_parameter'].get_param('account_peppol.edi.mode') == 'demo'
+            return self.edi_mode == 'demo' and self.proxy_type == 'peppol'
+        demo_param = self.env['ir.config_parameter'].get_param('account_peppol.edi.mode') == 'demo'
+        if args:
+            return demo_param and args[1]['proxy_type'] == 'peppol'
```

Comment:
That part is not very readable what `args` contains...
Here, this is mocking the method `def _make_request(self, url, params=False):`
So, `params` could be `False`.
So, if `args[1]` could be `False`.
So `args[1]['proxy_type']` could raise a traceback...?
Tell me if I wrongly understand that part :see_no_evil:

Reply:
you're absolutely right :see_no_evil: although it shouldn't really happen that we get to the point, where there is no edi user and no params are provided :slightly_smiling_face: 
fixed